server/datastore/mysql/policies.go

```diff
@@ -21,28 +22,30 @@ func (ds *Datastore) NewGlobalPolicy(queryID uint) (*fleet.Policy, error) {
 		return nil, errors.Wrap(err, "getting last id after inserting policy")
 	}
 
-	return ds.Policy(uint(lastIdInt64))
+	return policyDB(ctx, ds.writer, uint(lastIdInt64))
```

Comment:
This is the one change that is not just about passing `context.Context` down to the datastore layer - it fixes a possible issue if using a read replica.

Reply:
If we don't want to merge this PR for the release, we should port back this change.

Reply:
As discussed, I will open a separate PR with just this change.